tools/ci_connector_ops/ci_connector_ops/qa_checks.py

```diff
@@ -84,6 +84,39 @@ def check_connector_icon_is_available(connector: Connector) -> bool:
     """
     return connector.icon_path.exists()
 
+
+def read_all_files_in_directory(
+    directory: Path, ignored_directories: Optional[Set[str]] = None, ignored_filename_patterns: Optional[Set[str]] = None
+) -> Iterable[Tuple[str, str]]:
+    ignored_directories = ignored_directories if ignored_directories is not None else {}
+    ignored_filename_patterns = ignored_filename_patterns if ignored_filename_patterns is not None else {}
+
+    for path in directory.rglob("*"):
+        ignore_directory = any([ignored_directory in path.parts for ignored_directory in ignored_directories])
+        ignore_filename = any([path.match(ignored_filename_pattern) for ignored_filename_pattern in ignored_filename_patterns])
+        ignore = ignore_directory or ignore_filename
+        if path.is_file() and not ignore:
+            try:
+                for line in open(path, "r"):
+                    yield path, line
+            except UnicodeDecodeError:
+                continue
```

Comment:
Its a good idea to at least log something here

Reply:
Will do.